src/app/Library/Uploaders/Support/FileNameGenerator.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+namespace Backpack\CRUD\app\Library\Uploaders\Support;
+
+use Backpack\CRUD\app\Library\Uploaders\Support\Interfaces\FileNameGeneratorInterface;
+use Illuminate\Http\UploadedFile;
+use Illuminate\Support\Str;
+
+class FileNameGenerator implements FileNameGeneratorInterface
+{
+    public function getName(string|UploadedFile $file): string
```

Comment:
Hmmm could/should we make this `for()` to match the other one? And maybe it reads better? `FileNameGenerator::for('somefile.txt')` - what do you think @pxpm?

Reply:
I agree with this, let's dor it 💪